### DISCUSSION TITLE
Flexible  and responsive child component

### DISCUSSION THREAD

**Discussion by micst96** [quasarframework/quasar#16645] [General - Components / Directives / etc] [answered]
Hi, I was looking for solution to make component flexible and responsive - to occupy all available width and height.

I have found that I should set 'row' and 'col' for the class of i.e. div element to make it responsive and occupy whole space.

 Is it good approach? I couldn't find example/explanation for this solution and I am not sure why it works like that, why i.e. only 'col' does not allow to take all available space, but col + row does.

**Comment by pdanpdan** [comment]
https://quasar.dev/layout/grid/introduction-to-flexbox

**Reply by micst96** [reply]
Please read last sentence from 1st post. Front is new for me, I know this doc, but still there is no mention about situation where I can use  <div class="col row"</div>. Maybe it is trivial question, but looking on answers on the different forums, I couldn't find example with this solution, everyone is trying to solve it with raw CSS.

**Comment by pdanpdan** [comment, marked as answer]
.col alone takes the whole space for the element
.row creates a flex for the children of the element

Make a codepen with what you are doing if it's not working like this

**Reply by micst96** [reply]
ok, thanks for confirmation, I have found this explanation using GPT, but I was quite surprised that people ask about it on i.e. stack and there is no answers, but maybe it is such trivial thing that no one discuss about it :) thanks again.
